server/channels/app/integration_action.go

```diff
@@ -497,7 +497,13 @@ func (a *App) SubmitInteractiveDialog(c request.CTX, request model.SubmitDialogR
 	defer resp.Body.Close()
 
 	var response model.SubmitDialogResponse
-	json.NewDecoder(resp.Body).Decode(&response) // Don't fail, an empty response is acceptable
+	err = json.NewDecoder(resp.Body).Decode(&response) // Don't fail, an empty response is acceptable
+	if err != nil {
+		if errors.Is(err, io.EOF) {
+			return &response, nil
+		}
```

Comment:
Are there other cases where an EOF is returned that are not an empty response?

0/5 Should we just use `io.ReadAll` and check its length instead?

Reply:
Yeah there maybe some cases. Should i only use io.ReadAll for the empty response body check right. Because if i use readAll i can also use json.Unmarshal instead of decode. But the io.ReadAll method can affect some performance if the response body is too big as all of it gets load into memory.

Reply:
Yeah but io.ReadAll is better approach here as it will give true idea that the response body is empty.

Reply:
Or what i can do is read small portion of the response lets say 1KB using limited buffer and then check if its empty.

Reply:
What do you think @hanzei

Reply:
I think `io.ReadAll` is fine. The json.decode also reads the response body into memory, so that doesn't change.